pennylane/ops/op_math/controlled_ops.py

```diff
@@ -0,0 +1,124 @@
+
+    **Example**
+
+    The following shows how a single-qubit unitary can be applied to wire ``2`` with control on
+    both wires ``0`` and ``1``:
+
+    >>> U = np.array([[ 0.94877869,  0.31594146], [-0.31594146,  0.94877869]])
+    >>> qml.ControlledQubitUnitary(U, control_wires=[0, 1], wires=2)
+
+    Alternatively, the same operator can be constructed with a QubitUnitary:
+
+    >>> base = qml.QubitUnitary(U, wires=2)
+    >>> qml.ControlledQubitUnitary(base, control_wires=[0, 1])
+
+    Typically controlled operations apply a desired gate if the control qubits
+    are all in the state :math:`\vert 1\rangle`. However, there are some situations where
+    it is necessary to apply a gate conditioned on all qubits being in the
+    :math:`\vert 0\rangle` state, or a mix of the two.
+
+    The state on which to control can be changed by passing a string of bits to
+    `control_values`. For example, if we want to apply a single-qubit unitary to
+    wire ``3`` conditioned on three wires where the first is in state ``0``, the
+    second is in state ``1``, and the third in state ``1``, we can write:
+
+    >>> qml.ControlledQubitUnitary(U, control_wires=[0, 1, 2], wires=3, control_values=[0, 1, 1])
+
+    or
+
+    >>> qml.ControlledQubitUnitary(U, control_wires=[0, 1, 2], wires=3, control_values=[False, True, True])
+    """
+    num_wires = AnyWires
+    """int: Number of wires that the operator acts on."""
+
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = (2,)
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    grad_method = None
+    """Gradient computation method."""
+
+    # pylint: disable= too-many-arguments
+    def __init__(
+        self,
+        base,
+        control_wires=None,
+        wires=None,
+        control_values=None,
+        do_queue=True,
+        unitary_check=False,
+        work_wires=None,
+    ):
+        if getattr(base, "wires", False) and wires is not None:
+            warnings.warn(
+                "base operator already has wires; values specified through wires kwarg will be ignored."
+            )
+
+        if isinstance(base, Iterable):
+            base = QubitUnitary(base, wires=wires, unitary_check=unitary_check)
+
+        super().__init__(base, control_wires, control_values=control_values, do_queue=do_queue)
+        self._name = "ControlledQubitUnitary"
+
+    def _controlled(self, wire):
+        ctrl_wires = self.control_wires + wire
+        values = None if self.control_values is None else self.control_values + [True]
+        new_op = ControlledQubitUnitary(
+            self.base,
+            control_wires=ctrl_wires,
+            control_values=values,
+            work_wires=self.work_wires,
+        )
+        return new_op.inv() if self.inverse else new_op
```

Comment:
Given I'm trying to get rid of in-place inversion, we should just go ahead and get rid of it here and now.

```suggestion
        return new_op
```